drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -95,6 +100,43 @@ def ssn_serial
     self.SSN&.[](-4..-1)
   end
 
+  def mci_external_id
+    external_ids_by_slug('clientview').first
+  end
+
+  def warehouse_id
+    destination_client.id
+  end
+
+  def warehouse_url
+    "https://#{ENV['FQDN']}/clients/#{id}/from_source"
+  end
+
+  def mci_url
+    return unless mci_external_id&.remote_credential&.present?
```

Comment:
The RemoteCredential for the MCI ID will be different, so you'll need to look for a `clientview` credential explicitly here. (The `mci` remote credential [is used](https://github.com/greenriver/hmis-warehouse/blob/2916a0c5e1fd24a1901c9dde70367a66153f76b6/drivers/hmis_external_apis/app/models/hmis_external_apis/mci.rb#L10) for storing oauth information thats used to create/link the MCI IDs.)

There shouldn't be multiple `clientview` credentials, so you can just do `RemoteCredential.where(slug: "clientview").first&.link_base`,

Reply:
Can you also throw in:
```
return unless client.enrollments.viewable_by(current_user).exists?
```

The ClientView URL is only supposed to be shown if the User has access to see any of the Client's enrollments. We may need to tweak that logic later (to be only current/recent enrollments) but thats a start.